src/Components/Article.jsx

```diff
@@ -27,29 +25,34 @@ function Article({ articles }) {
   }, [article_id]);
 
   useEffect(() => {
-    fetch(
-      `https://nc-news-y5oc.onrender.com/api/articles/${article_id}/comments`
-    )
-      .then((comments) => {
-        return comments.json();
-      })
-      .then((comments) => {
-        setComments(comments);
-      });
-  }, []);
+    getCommentsById(article_id).then((comments) => {
+      setComments(comments);
+    });
+  }, [article_id]);
 
-  useEffect(() => {
-    let totalVotes = 0;
-    comments.map((comment) => {
-      return (totalVotes += comment.votes);
+  function incrLike() {
+    upvote(article_id).then(({ data }) => {
+      setArticle(data)
+    });
+  }
+
+  function decrLike() {
+    downvote(article_id).then(({ data }) => {
+      setArticle(data)
```

Comment:
Ideally we should have one function here that can handle an upvote or a down vote. They the voting should also be handled optimistically (and then removed if the API patch request fails) Have a look at [these notes](https://l2c.northcoders.com/courses/sd-notes/front-end#sectionId=optimistic-rendering,step=intro) for a reminder.

As you are using the response from the api call to update the votes what happens, in theory, if other users vote too. The increase or decrease amount wouldn't make sense for the users on click.

It might be a good idea to disable the button so a user and only vote to increase/decrease by one. This is not as important as the above though